src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -265,36 +283,65 @@ fn compute_deps(
         // If we've gotten past all that, then this dependency is
         // actually used!
         true
-    });
+    };
+    let filtered_deps = state.deps(unit, unit_for, dep_filter);
 
     let mut ret = Vec::new();
     let mut dev_deps = Vec::new();
-    for (id, deps) in filtered_deps {
-        let pkg = state.get(id);
-        let lib = match pkg.targets().iter().find(|t| t.is_lib()) {
+    for (dep_pkg_id, deps) in filtered_deps {
+        let dep_pkg = state.get(dep_pkg_id);
+        // Artifact dependencies are only counted as standard libraries if they are marked
```

Comment:
"standard libraries" sounds like you are referring to STD. Is that what is meant here?

Reply:
A good point - while trying to rephrase the sentence I realized that by now the code has moved elsewhere, making the comment misleading.

This is fixed in 09b51910e3e23330b10a0a7c295c775652ec3b15 .